speckle/Cache.py

```diff
@@ -0,0 +1,169 @@
+import sqlite3, contextlib
+from urllib.request import pathname2url
+
+import os
+
+class SpeckleCache():
```

Comment:
Neat new feature 😄 Can I convince you to write tests for these so that if we make changes later we're sure to not break things? 

Also worth adding some documentation for this no?

Reply:
Yes, this is quite preliminary, I guess a bit too eager to get it out the door. I'll try to make some time to write some tests + docs ;)

Let's hold off on this PR for now then.

Reply:
Well we don't have to hold it off. You can take your time to write tests and once you're ready just ping me to review it again 😄